gcs_to_bq/main.py

```diff
@@ -0,0 +1,42 @@
+import base64
+import json
+import logging
+from common.census_to_bq import write_state_names_to_bq
+
+
+_HOUSEHOLD_INCOME = 'HOUSEHOLD_INCOME'
+_STATE_NAMES = 'STATE_NAMES'
+
+
+def ingest_bucket_to_bq(event, context):
+  """Main GCF function for moving data from buckets to bigquery.
+     Triggered by notify-data-ingested topic.
+
+     event: Dict containing the Pub/Sub method. The Cloud Scheduler payload will be a base-64
+            encoded string in the 'data' field.
+     context: Metadata about this function invocation."""
+  # TODO: how to share code between functions? Would be nice to abstract some of this
+  # message processing boilerplate
+  try:
+    if 'data' not in event:
+      logging.warning("PubSub message missing 'data' field")
+      return
+    data = base64.b64decode(event['data']).decode('utf-8')
+    logging.info('Message received: {}'.format(data))
+  except json.JSONDecodeError as err:
+    logging.error("Could not load json object: %s", err)
```

Comment:
nit: but we could create a `util.py` file that could do these like decode pub/subs.

Reply:
So..... I don't know how to share code between cloud functions haha. From a brief internet search there isn't a simple way to do it. Maybe it's possible if you create a separate package and then install the package via requirements.txt? Jennie had also suggested using terraform to copy shared code to different functions.

@slagathorr do you know anything about this?

Reply:
Ah you can't have code in one GCF read from another GCF. The only way I can think of is having the dependent code in a shared file while developing, and then when that changes it would repack and redeploy all the functions that use that. There's likely a different (and better) way in Cloud Run to do that, so I wouldn't worry too much about it not being easy in GCF. @mayaspivak may have some thoughts on how to do that in GCR, but don't worry too much about it in GCF for the prototype.

Reply:
Okay, for now we can just copy/paste if there's shared code, and hopefully Cloud Run will smooth things out

Reply:
Since we control the build for cloud run, docker can do the copying for us in the docker build. (which is really not all that different than terraform doing it). There is server-side deduplication in cloud run for sure and probably in the container registry as well. 

This will allow us to have only one copy of the shared code in our repository. But as Brian mentioned above and I previously mentioned in the Devops Plan doc, we need to make sure each service is redeployed whenever the shared code is updated. Which we should be able to detect more easily in cloud run using the container digest.

Reply:
Wait, if we create a top level directory (at the same level as `gcs_to_bq` and `data_ingestion`) and put this `util.py` file in it, shouldn't it be importable and packaged with both functions when we deploy?

Reply:
Makes sense, thanks for the context!

Reply:
Shoot sorry Lorenzo, I keep having race conditions with my responses

I think the issue is that the way functions are deployed requires a certain structure, where the main.py file is supposed to be at the root directory, and it uploads the contents of that directory. I don't think it can reference a parent or sibling directory. I'm not 100% sure about this, I haven't actually tried it, but from doing a bit of reading it doesn't seem like it'll work. May be worth trying though :)